src/app/util/ClientMonitoringRegistrationTable.h

```diff
@@ -19,33 +19,38 @@
 
 #include <app-common/zap-generated/cluster-objects.h>
 #include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/support/CodeUtils.h>
-#include <platform/CHIPDeviceConfig.h>
 
 namespace chip {
 class ClientMonitoringRegistrationTable
 {
 public:
     using MonitoringRegistrationStruct = chip::app::Clusters::ClientMonitoring::Structs::MonitoringRegistration::Type;
 
-    ClientMonitoringRegistrationTable(FabricIndex fabricIndex);
+    ClientMonitoringRegistrationTable(PersistentStorageDelegate & storage);
     ~ClientMonitoringRegistrationTable(){};
 
-    void SaveToStorage();
+    CHIP_ERROR SaveToStorage();
+    CHIP_ERROR LoadFromStorage(FabricIndex fabricIndex);
 
     // Getter
     NodeId getClientNodeId();
     uint64_t getICid();
     FabricIndex getFaricIndex();
+    MonitoringRegistrationStruct getRegisteredClient();
 
     // Setter
     void setClientNodeId(NodeId clientNodeId);
     void setICid(uint64_t ICid);
     void setFabricIndex(FabricIndex fabric);
 
 private:
-    void LoadFromStorage(FabricIndex fabricIndex);
+    static constexpr uint8_t kRegStorageSize = TLV::EstimateStructOverhead(sizeof(NodeId), sizeof(uint64_t), sizeof(FabricIndex));
+
+    CHIP_ERROR IsRegisteredClientValid();
     MonitoringRegistrationStruct mRegisteredClient;
```

Comment:
Why just one?  Shouldn't there be one per fabric?

Reply:
The idea is each instance of the `ClientMonitoringRegistrationTable` can only load and store one fabric at a time.
It's read / write one fabric at a time.